MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -119,6 +127,11 @@ public string GetVideoEncoder(EncodingJobInfo state, EncodingOptions encodingOpt
 
             if (!string.IsNullOrEmpty(codec))
             {
+                if (string.Equals(codec, "h265", StringComparison.OrdinalIgnoreCase) ||
```

Comment:
All these strings, `h265`, `hevc`, `265` and all the others used through out the code, should probably be enums or something.

Reply:
Absolutely agree. I've never seen so many string comparisons in a chunk of code in my life. I initially went down the road of creating enums, but I quickly realized how much code this was going to touch so I changed my mind. I believe that change should go in its own Issue and PR.

Reply:
I'm gonna give you a little tip... enums are a bad idea...  Take it or leave it.

Reply:
Can you elaborate? It seems like codecs for example internally in ffmpeg are already essentially an enumeration, besides they do not change all that often. And for serialization you can specify the string values if you do not want to deal with numbers (codec IDs for example) in your JSON bodies.